### PR TITLE
Fix saving coordinators in CoordinatorClusterState

### PR DESCRIPTION
**Pull request by as51340**
### Description

Fix handling of coordinators in `CoordinatorClusterState`.

[master < Task] PR
- [x] Provide the full content or a guide for the final git message
    - **Fix coordinators in `CoordClusterState`**

### Documentation checklist
- [x] Add the documentation label tag
- [x] Add the bug / feature label tag
- [x] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [ ] Write a release note, including added/changed clauses
    - **[Release note text]**
- [ ] Link the documentation PR here
    - **[Documentation PR link]**
- [x] Tag someone from docs team in the comments

### CODE REVIEW

**Comment by as51340** [comment]
@kgolubic just a bug fix.. This wasn't even released yet so no problem

File: tests/e2e/high_availability/distributed_coords.py [hunk 1]
```diff
@@ -1927,207 +1927,207 @@ def get_vertex_count():
...
-
-        return get_vertex_count
-
-    instance_1_cursor = connect(port=7687, host="localhost").cursor()
-    instance_2_cursor = connect(port=7688, host="localhost").cursor()
-
-    mg_sleep_and_assert(vertex_count, get_vertex_count_func(instance_1_cursor))
-    mg_sleep_and_assert(vertex_count, get_vertex_count_func(instance_2_cursor))
-    mg_sleep_and_assert(vertex_count, get_vertex_count_func(instance_3_cursor))
+# def test_force_reset_works_after_failed_registration_and_2_coordinators_down():
```
Comment (as51340) [inline tests/e2e/high_availability/distributed_coords.py:1930]: @antoniofilipovic will fix it when he comes back, he was working on it

File: tests/unit/coordinator_instance.cpp [hunk 2]
```diff
@@ -52,11 +49,11 @@ class CoordinatorInstanceTest : public ::testing::Test {
 //
 //   auto const coord_to_replica_config =
```
Comment (as51340) [inline tests/unit/coordinator_instance.cpp:50]: @antoniofilipovic already fixed this but I made a mistake with rebase, he will revert when comes back

File: src/coordination/coordinator_instance.cpp [hunk 3]
```diff
@@ -95,7 +95,8 @@ CoordinatorInstance::CoordinatorInstance(CoordinatorInstanceInitConfig const &co
               repl_instances_.clear();
               spdlog::info("Stopped all replication instance frequent checks.");
             });
-          })) {
+          })),
+      config_{config} {
```
Comment (andrejtonev) [inline src/coordination/coordinator_instance.cpp:99]: In general this constructor is very hard to read. Could we break it down somehow?

File: src/coordination/coordinator_instance.cpp [hunk 4]
```diff
@@ -573,6 +575,16 @@ auto CoordinatorInstance::RegisterReplicationInstance(CoordinatorToReplicaConfig
...
 
+  if (!raft_state_.CoordinatorExists(config_.coordinator_id)) {
+    auto const self_c2c_config = CoordinatorToCoordinatorConfig{
+        .coordinator_id = config_.coordinator_id,
+        .bolt_server = io::network::Endpoint{"0.0.0.0", static_cast<uint16_t>(config_.bolt_port)},
+        .coordinator_server = io::network::Endpoint{"0.0.0.0", static_cast<uint16_t>(config_.coordinator_port)}};
+    if (!raft_state_.AppendAddCoordinatorInstanceLog(self_c2c_config)) {
+      spdlog::error("Failed to append add coordinator instance log for self!");
+    }
+  }
```
Comment (andrejtonev) [inline src/coordination/coordinator_instance.cpp:586]: Why IP 0.0.0.0?

File: src/coordination/include/coordination/instance_status.hpp [hunk 5]
```diff
@@ -27,6 +27,12 @@ struct InstanceStatus {
   std::string coord_socket_address;
   std::string cluster_role;
   std::string health;
+
+  friend bool operator==(InstanceStatus const &lhs, InstanceStatus const &rhs) {
+    return lhs.instance_name == rhs.instance_name && lhs.raft_socket_address == rhs.raft_socket_address &&
+           lhs.coord_socket_address == rhs.coord_socket_address && lhs.cluster_role == rhs.cluster_role &&
+           lhs.health == rhs.health;
+  }
```
Comment (andrejtonev) [inline src/coordination/include/coordination/instance_status.hpp:35]: Would default do the trick?

File: src/coordination/include/nuraft/coordinator_cluster_state.hpp [hunk 6]
```diff
@@ -77,7 +77,7 @@ void from_json(nlohmann::json const &j, CoordinatorInstanceState &instance_state
 // Source of truth since it is modified only as the result of RAFT's commiting
 class CoordinatorClusterState {
  public:
-  explicit CoordinatorClusterState(CoordinatorInstanceInitConfig const &config);
+  CoordinatorClusterState() = default;
```
Comment (andrejtonev) [inline src/coordination/include/nuraft/coordinator_cluster_state.hpp:80]: The config is now saved in the instance, right?

**Review by andrejtonev — APPROVED** [review]
Looks good

File: src/coordination/include/nuraft/coordinator_cluster_state.hpp (hunk 6 above)
Comment (as51340) [reply src/coordination/include/nuraft/coordinator_cluster_state.hpp:80]: yes

File: src/coordination/include/coordination/instance_status.hpp (hunk 5 above)
Comment (as51340) [reply src/coordination/include/coordination/instance_status.hpp:35]: I think so

File: src/coordination/coordinator_instance.cpp (hunk 4 above)
Comment (as51340) [reply src/coordination/coordinator_instance.cpp:586]: this is for adding myself. 0.0.0.0 is better than 127.0.0.1. This is the only reason

File: src/coordination/coordinator_instance.cpp (hunk 3 above)
Comment (as51340) [reply src/coordination/coordinator_instance.cpp:99]: Agree, will think about it

**Comment by as51340** [comment]
@kgolubic changes from this PR are overwritten by #1891 . No need to do anything here

**Pull request merged**